common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -1812,6 +1812,24 @@ private static Geometry[] convertGeometryToArray(Geometry geom) {
     return array;
   }
 
+  public static Geometry generatePoints(Geometry geom, int numPoints, long seed) {
+    RandomPointsBuilderSeed pointsBuilder = new RandomPointsBuilderSeed(geom.getFactory(), seed);
+    pointsBuilder.setExtent(geom);
+    pointsBuilder.setNumPoints(numPoints);
+    return pointsBuilder.getGeometry();
+  }
+
+  public static Geometry generatePoints(Geometry geom, int numPoints) {
+    return generatePoints(geom, numPoints, 0);
+  }
+
+  public static Geometry generatePoints(Geometry geom, int numPoints, Random random) {
```

Comment:
Why is there another overloaded function that takes `Random` as input?

Reply:
This is for using the `random` member of `ST_GeneratePoints` to generate consistent random streams